### PR TITLE
Arbab/change name coursediscovery

### PR DESCRIPTION
**Pull request by arbabnazar**
@edx/devops kindly review, this change is linked with [this](https://github.com/edx-ops/edx-secure/pull/700).
@cpennington @clintonb kindly have a look, if I have missed something to rename.

Thanks

### CODE REVIEW

**Comment by maxrothman** [comment]
@arbabnazar the travis build is failing at collectstatic. Did you pass a name change in where you shouldn't have?

File: docker/build/discovery/ansible_overrides.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+discovery_gunicorn_host: 0.0.0.0
+DISCOVERY_MYSQL: 'db'
+DISCOVERY_DJANGO_SETTINGS_MODULE: 'discovery.settings.devstack'
```
Comment (cpennington) [inline docker/build/discovery/ansible_overrides.yml]: Are there corresponding changes in the course_discovery repo?

Comment (maxrothman) [inline docker/build/discovery/ansible_overrides.yml]: @cpennington AFAIK there shouldn't be. @arbabnazar this line shouldn't have been changed.

**Comment by arbabnazar** [comment]
@cpennington @clintonb @e0d test fail on this while collectstatic:

```
stderr: Traceback (most recent call last):
  File "/edx/app/discovery/discovery/course_discovery/settings/utils.py", line 11, in get_env_setting
    return environ[setting]
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/os.py", line 683, in __getitem__
    raise KeyError(key) from None
KeyError: 'COURSE_DISCOVERY_CFG'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "manage.py", line 15, in <module>
    execute_from_command_line(sys.argv)
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/site-packages/django/core/management/__init__.py", line 182, in fetch_command
    settings.INSTALLED_APPS
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/site-packages/django/conf/__init__.py", line 48, in __getattr__
    self._setup(name)
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/site-packages/django/conf/__init__.py", line 44, in _setup
    self._wrapped = Settings(settings_module)
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/site-packages/django/conf/__init__.py", line 92, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/edx/app/discovery/venvs/discovery/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 662, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/edx/app/discovery/discovery/course_discovery/settings/devstack.py", line 1, in <module>
    from course_discovery.settings.production import *
  File "/edx/app/discovery/discovery/course_discovery/settings/production.py", line 37, in <module>
    CONFIG_FILE = get_env_setting('COURSE_DISCOVERY_CFG')
  File "/edx/app/discovery/discovery/course_discovery/settings/utils.py", line 14, in get_env_setting
    raise ImproperlyConfigured(error_msg)
django.core.exceptions.ImproperlyConfigured: Set the [COURSE_DISCOVERY_CFG] env variable!
make: *** [static] Error 1
stdout: python manage.py collectstatic --noinput
```

**Comment by cpennington** [comment]
You'll need to either update the `course_discovery` code to look for either environment variable, or not change the `COURSE_DISCOVERY_CFG` environment variable name in the `configuration` repo.

**Comment by maxrothman** [comment]
@arbabnazar When I last talked to the team, we agreed that the course discovery app will continue to refer to itself as `course_discovery` (in repo name, environment variable names, etc.) Therefore, anywhere where the role touches the actual code (e.g. in @cpennington's comment above), it should remain `course_discovery`.

Also, you should rebase this branch to resolve the merge conflicts.

**Comment by maxrothman** [comment]
Why was this PR closed?

**Comment by feanil** [comment]
A different version of this PR was merged a while ago by Ed.

**Comment by maxrothman** [comment]
Looks like it hasn't yet been merged. Leaving this link here for future reference: https://github.com/edx/configuration/pull/2691

**Pull request closed without merging**